src/main/java/org/fogbowcloud/arrebol/execution/docker/DockerCommandExecutor.java

```diff
@@ -0,0 +1,57 @@
+package org.fogbowcloud.arrebol.execution.docker;
+
+import org.apache.log4j.Logger;
+import org.fogbowcloud.arrebol.execution.docker.request.ExecInstanceResult;
+import org.fogbowcloud.arrebol.execution.docker.request.WorkerDockerRequestHelper;
+
+import static java.lang.Thread.sleep;
+
+public class DockerCommandExecutor {
+
+    private static final Logger LOGGER = Logger.getLogger(DockerCommandExecutor.class);
+    private final WorkerDockerRequestHelper workerDockerRequestHelper;
+
+    public DockerCommandExecutor(WorkerDockerRequestHelper workerDockerRequestHelper) {
+        this.workerDockerRequestHelper = workerDockerRequestHelper;
+    }
+
+    public Integer executeCommand(String command, String taskId) throws Exception {
+        LOGGER.info("Sending command to the [" + command + "] for the task [" + taskId + "]"
+                + this.workerDockerRequestHelper.getContainerName() + "].");
+
+        ExecInstanceResult execInstanceResult = executeCommand(command);
+
+        LOGGER.info("Executed command [" + command + "] for the task [" + taskId
+                + "] with exitcode=[" + execInstanceResult.getExitCode() + "] in worker ["
+                + this.workerDockerRequestHelper.getContainerName() + "].");
+        return execInstanceResult.getExitCode();
+    }
+
+    private ExecInstanceResult executeCommand(String command) throws Exception {
+        String execId = this.workerDockerRequestHelper.createExecInstance(command, false, false);
+        this.workerDockerRequestHelper.startExecInstance(execId);
+
+        ExecInstanceResult execInstanceResult =
+                this.workerDockerRequestHelper.inspectExecInstance(execId);
+        while (execInstanceResult.getExitCode() == null) {
+            execInstanceResult = this.workerDockerRequestHelper.inspectExecInstance(execId);
+            final long poolingPeriodTime = 300;
```

Comment:
how about moving this to the the docker config file? is it worthy doing? at least, declare as a private final static in the class.

Reply:
I declared as a private final static in the class.